client/src/pages/SignUpModal.tsx

```diff
@@ -0,0 +1,201 @@
+import './Modal.scss';
+import { Button, OverlayTrigger, Popover } from 'react-bootstrap';
+import React, { useState } from 'react';
+import CommonModal from '../components/organisms/Modal';
+import { createNewAccount } from '../services/auth';
+import { Redirect } from 'react-router-dom';
+import { useApolloClient } from "@apollo/client";
+import { useDispatch } from "react-redux";
+
+function SignUpModal() {
+    const client = useApolloClient();
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [firstName, setFirstName] = useState("");
+    const [lastName, setLastName] = useState("");
+    const [hasOneLowerCase, setHasOneLowerCase] = useState(false);
+    const [hasOneUpperCase, setHasOneUpperCase] = useState(false);
+    const [hasOneNumber, setHasOneNumber] = useState(false);
+    const [hasOneSymbol, setHasOneSymbol] = useState(false);
+    const [hasTwelveCharacterMin, setHasTwelveCharacterMin] = useState(false);
+    const oneLowerCase= /^(?=.*[a-z])/;
+    const oneUpperCase=/^(?=.*[A-Z])/;
+    const oneNumber=/^(?=.*[0-9])/;
+    const oneSymbol=/^(?=.*[*!@#$%^&(){}[\]:;<>,.?/~_+\-=|\\])/;
+    const twelveCharacterMin=/^(?=.{12,})/;
+    const initialReq : string[] = ["at least 1 lowercase letter", "at least 1 uppercase letter","at least 1 number","at least 1 symbol","12 characters minimum" ]
+    const [requirements, setRequirements] = useState(initialReq);
+    const [show, setShow] = useState(true);
+    const [redirectToHome, setRedirectToHome] = useState(false);
+
+    const handleClose = () => setShow(false);
+    const handleShow = () => setShow(true);
+
+    const handleClick = async (e:any) => { 
+        e.preventDefault();
+        createNewAccount(email, password);
+        setRedirectToHome(true);
+    };
+
+    const onChangeFirstName = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setFirstName(e.target.value);
+    };
+
+    const onChangeLastName = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setLastName(e.target.value);
+    }
+
+    const onChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setEmail(e.target.value);
+    };
+
+    const onChangePass = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const password = e.target.value;
+        const copy:string[]= requirements;
+        // checks each of the password requirements and updates the array
+        if (!oneLowerCase.test(password)) {
```

Comment:
I think we can really shorten the code for `onChangePass`
Let me know if the following code outline makes sense

```
const requirementToStateSetterMap = new Map ([
  ['lowerCase', /^(?=.*[a-z])/],
  ['uppserCase', /^(?=.*[A-Z])/],
...
])

const requirementToTestMap = new Map ([
  ['lowerCase', setHasOneLowerCase],
  ['uppserCase', setHasOneUppCase],
...
])

const requirementToMessageMap = new Map ([
  ['lowerCase', "at least 1 lowercase letter"],
  ['uppserCase', "at least 1 uppercase letter"],
...
])


const onChangePass = (e: React.ChangeEvent<HTMLInputElement>) => {
   const password = e.target.value;
   const copy:string[]= requirements;

  requirementToTestMap.forEach((test, key) => {
    const result = test.test(password)
    const message = requirementToMessageMap[key]

    requirementToStateSetterMap[key](result)
    if(result && !copy.includes(message)){
        copy.push(message);
    }
    else if(!result && copy.includes(message)){
        const index = copy.indexOf(message);
        copy.splice(index, 1); 
    }

    setRequirements(copy);
  })
}
```